crates/nargo_cli/src/cli/check_cmd.rs

```diff
@@ -1,53 +1,56 @@
-use crate::{errors::CliError, resolver::resolve_root_manifest};
+use crate::{errors::CliError, manifest::resolve_workspace_in_directory, prepare_package};
 use acvm::Backend;
 use clap::Args;
 use iter_extended::btree_map;
+use nargo::package::Package;
 use noirc_abi::{AbiParameter, AbiType, MAIN_RETURN_NAME};
 use noirc_driver::{check_crate, compute_function_signature, CompileOptions};
 use noirc_errors::reporter::ReportedErrors;
-use noirc_frontend::{graph::CrateId, hir::Context};
-use std::path::{Path, PathBuf};
+use noirc_frontend::{
+    graph::{CrateId, CrateName},
+    hir::Context,
+};
 
 use super::fs::write_to_file;
 use super::NargoConfig;
-use crate::constants::{PROVER_INPUT_FILE, VERIFIER_INPUT_FILE};
 
 /// Checks the constraint system for errors
 #[derive(Debug, Clone, Args)]
 pub(crate) struct CheckCommand {
+    /// The name of the package to check
+    #[clap(long)]
+    package: Option<CrateName>,
+
     #[clap(flatten)]
     compile_options: CompileOptions,
 }
 
 pub(crate) fn run<B: Backend>(
-    backend: &B,
+    _backend: &B,
     args: CheckCommand,
     config: NargoConfig,
 ) -> Result<(), CliError<B>> {
-    check_from_path(backend, &config.program_dir, &args.compile_options)?;
-    println!("Constraint system successfully built!");
+    let workspace = resolve_workspace_in_directory(&config.program_dir, args.package)?;
+
+    for package in &workspace {
+        check_package(package, &args.compile_options)?;
+        // TODO: Print package name at beginning?
+        println!("Constraint system successfully built!");
```

Comment:
If we're printing multiple times for each package then we should prefix this.

Reply:
Done